### ISSUE TITLE
Connect to other nodes in the cluster when starting a node automatically?

### ISSUE THREAD

**Issue by smartepsh** [sindrip/ex_unit_cluster#1]
Thank you very much for this dependency. It provides great convenience for writing cluster tests.

Can you provide an option for `ExUnitCluster.start_node/2` to control the behaviour that connect to the other nodes in same cluster automatically or not? (Even do this at the very beginning of the boot process).

## Example 

- Scenario 1
**Test a GenServer using the `handle_continue/2` initialization state by getting data from other nodes in the cluster.**

Currently, a node started by `ExUnitCluster.start_node/2` will first complete the startup, then connect to the existing nodes by calling `ExUnitCluster.call(cluster, new_node, Node, :connect, [base_node])` manually. 

So I would miss the opportunity to test the GenServer init process. What I need here is to connect to the cluster first then start the GenServer.

- Scenario 2

**Test a GenServer with send sync signal periodically, especially the moment that join the cluster.**

But for this one, I just want to control the connection manually.

**Comment by sindrip** [comment]
Hi @smartepsh,

Can you try out the changes in #2 and let me know if they cover your scenarios?

I've added a new option `join`
```elixir
node = ExUnitCluster.start_node(cluster, join: false)
```

Note that there is another option `applications` if you don't want to automatically start your application on node startup, so you might be able to do something like:
```elixir
node = ExUnitCluster.start_node(cluster, applications: [], join: false)
in_cluster cluster, node do
  Application.ensure_all_started(:my_app)
  ...
end
```

If this works, I can merge the PR and release a new version for you.

**Comment by smartepsh** [comment]
@sindrip I've tried it, works well. 

I'm really looking forward to this release. :joy:

**Comment by sindrip** [comment]
I've released version 0.4.0 with these changes, it is available in Hex :smile:
